src/PIL/Jpeg2KImagePlugin.py

```diff
@@ -154,9 +154,6 @@ def _parse_jp2_header(fp):
             if reader.read_fields(">4s")[0] == b"jpx ":
                 mimetype = "image/jpx"
 
-    if header is None:
-        raise SyntaxError("Could not find JP2 header")
```

Comment:
You're right, this is unreachable, `reader.has_next_box()` will never return False (because `reader = BoxReader(fp)` is initialized without a length), so the only possible outcome for a malformed image not containing a "jp2h" header is that `reader.next_box_type()` will fail with some kind of exception.

While not exactly elegant, that's definitely better than having an if statement that misleads the reader. 👍